src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -881,14 +902,9 @@ static internal object UnsafeGetUninitializedObject(int id)
         static internal object TryGetUninitializedObjectWithFormatterServices(Type type)
         {
             object obj = null;
-            var formatterServiceType = typeof(string).GetTypeInfo().Assembly.GetType("System.Runtime.Serialization.FormatterServices");
-            if (formatterServiceType != null)
+            if (s_getUninitializedObjectDelegate != null)
             {
-                var methodInfo = formatterServiceType.GetMethod("GetUninitializedObject", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static);
-                if (methodInfo != null)
-                {
-                    obj = methodInfo.Invoke(null, new object[] { type });
-                }
+                obj = s_getUninitializedObjectDelegate(type);
```

Comment:
Sorry for not pointing this out earlier, but this whole method could be simplified as:

```csharp
static internal object TryGetUninitializedObjectWithFormatterServices(Type type) =>
    s_getUninitializedObjectDelegate?.Invoke(type);
```